exp/api/v1beta1/azuremachinepool_default_test.go

```diff
@@ -174,6 +202,45 @@ func TestAzureMachinePool_SetDiagnosticsDefaults(t *testing.T) {
 	g.Expect(nilDiagnostics.machinePool.Spec.Template.Diagnostics.Boot.StorageAccountType).To(Equal(infrav1.ManagedDiagnosticsStorage))
 }
 
+func TestAzureMachinePool_SetSpotEvictionPolicyDefaults(t *testing.T) {
+	g := NewWithT(t)
+
+	type test struct {
+		machinePool *AzureMachinePool
+	}
+
+	// test for SetSpotEvictionPolicyDefaults() to Ensure the default policy is being set correctly depending on the input1
+	defaultPolicy := infrav1.SpotEvictionPolicyDeallocate
```

Comment:
I don't think this needs to be set to its own local variable. Just using `infrav1.SpotEvictionPolicyDeallocate` is fine as it is a constant.

Reply:
I have checked that earlier. If I dont use local vairable than it gives error as shown below : 
![image](https://user-images.githubusercontent.com/83772913/234610334-38fd276a-e4cd-433a-b706-6540c153901f.png)

Similarly it gives same error for "infrav1.SpotEvictionPolicyDelete". That's why I used local variables.